libs/vm-modules/src/core/byte_array_wrapper.cpp

```diff
@@ -42,16 +60,160 @@ ByteArrayWrapper::ByteArrayWrapper(VM *vm, TypeId type_id, byte_array::ByteArray
 
 Ptr<ByteArrayWrapper> ByteArrayWrapper::Constructor(VM *vm, TypeId type_id, int32_t n)
 {
-  return Ptr<ByteArrayWrapper>{
-      new ByteArrayWrapper(vm, type_id, byte_array::ByteArray(std::size_t(n)))};
+  return Ptr<ByteArrayWrapper>{new ByteArrayWrapper(vm, type_id, ByteArray(std::size_t(n)))};
 }
 
 Ptr<ByteArrayWrapper> ByteArrayWrapper::Copy()
 {
   return vm_->CreateNewObject<ByteArrayWrapper>(byte_array_.Copy());
 }
 
-byte_array::ByteArray ByteArrayWrapper::byte_array() const
+Ptr<vm::String> ByteArrayWrapper::ToBase64()
+{
+  return Ptr<vm::String>{new String{vm_, static_cast<std::string>(byte_array_.ToBase64())}};
+}
+
+bool ByteArrayWrapper::FromBase64(fetch::vm::Ptr<vm::String> const &value_b64)
+{
+  if (!value_b64)
+  {
+    return false;
+  }
+
+  try
+  {
+    auto decoded_array{ByteArray{value_b64->string()}.FromBase64()};
+
+    if (value_b64->Length() > 0 && decoded_array.empty())
+    {
+      return false;
+    }
+
+    byte_array_ = std::move(decoded_array);
+    return true;
+  }
+  catch (std::runtime_error const &ex)
+  {
+    return false;
+  }
+}
+
+fetch::vm::Ptr<vm::String> ByteArrayWrapper::ToHex()
+{
+  return Ptr<vm::String>{new String{vm_, static_cast<std::string>(byte_array_.ToHex())}};
+}
+
+bool ByteArrayWrapper::FromHex(fetch::vm::Ptr<vm::String> const &value_hex)
+{
+  if (!value_hex)
+  {
+    return false;
+  }
+
+  try
+  {
+    auto decoded_array{ByteArray{value_hex->string()}.FromHex()};
+
+    if (value_hex->Length() > 0 && decoded_array.empty())
+    {
+      return false;
+    }
+
+    byte_array_ = std::move(decoded_array);
+    return true;
+  }
+  catch (std::runtime_error const &ex)
+  {
+    return false;
+  }
+}
+
+fetch::vm::Ptr<vm::String> ByteArrayWrapper::ToBase58()
```

Comment:
I don't think byte array wrapper should have a base58 representation. I want to reserve this to mean addresses only